src/popup/views/SignTransaction/index.tsx

```diff
@@ -51,58 +103,155 @@ const SignTransaction = () => {
+                  <div>
+                    <strong>Amount</strong>
+                  </div>
+                  <div>
+                    {formattedAmount} {asset.code}
+                  </div>
+                </li>
+              ) : null}
+
+              {destination ? (
+                <li>
+                  <div>
+                    <strong>Destination</strong>
+                  </div>
+                  <div>{truncatedPublicKey(destination)}</div>
+                </li>
+              ) : null}
+
+              {signer ? (
+                <>
+                  <li>
+                    <div>
+                      <strong>Signer</strong>
+                    </div>
+                    <div>{truncatedPublicKey(signer.ed25519PublicKey)}</div>
+                  </li>
+                  <li>
+                    <div>
+                      <strong>Weight</strong>
+                    </div>
+                    <div>{signer.weight}</div>
+                  </li>
+                </>
+              ) : null}
+
+              {buying ? (
+                <li>
+                  <div>
+                    <strong>Buying</strong>
+                  </div>
+                  <div> {buying.code}</div>
+                </li>
+              ) : null}
+
+              {selling ? (
+                <li>
+                  <div>
+                    <strong>Selling</strong>
+                  </div>
+                  <div>{selling.code}</div>
+                </li>
+              ) : null}
+
+              {buyAmount ? (
+                <li>
+                  <div>
+                    <strong>Amount</strong>
+                  </div>
+                  <div>{formattedBuyAmount}</div>
+                </li>
+              ) : null}
+
+              {price ? (
+                <li>
+                  <div>
+                    <strong>Price</strong>
+                  </div>
+                  <div> {price}</div>
+                </li>
+              ) : null}
+            </ListEl>
+          </OperationBox>
+        );
+      },
     );
 
   return (
-    <>
-      <h1>Confirm Transaction</h1>
-      <img alt="favicon of site" src={`${url}/favico.png`} />
-
-      <h3>{title} is requesting a transaction signature</h3>
-      <h3>Transaction details</h3>
-      <p>Source Account Key: {publicKey}</p>
-      {fee ? <p>Base fee: {fee}</p> : null}
-      <h3>{_operations.length} Operations:</h3>
+    <El>
+      <BackButton onClick={() => window.location.replace("/")} />
+      <Header>Confirm Transaction</Header>
+      <Subheader>{title} is requesting a transaction</Subheader>
+      <ListEl>
+        <li>
+          <div>
+            <strong>Source Acc Key</strong>
+          </div>
+          <div>{truncatedPublicKey(publicKey)}</div>
+        </li>
+        {_fee ? (
+          <li>
+            <div>
+              <strong>Base fee</strong>
+            </div>
+            <div> {_fee}</div>
+          </li>
+        ) : null}
```

Comment:
since a lot fo these are constructed as
```
<li>
<div><strong>{title}</strong></div>
<div>{value}</div>
</li>
```

i wonder if it's worth creating a separate component for these 🤔

Reply:
Updated